src/SchedulerSubsystem/Scheduler.java

```diff
@@ -8,30 +8,57 @@
 import Networking.Transmitters.DMA_Transmitter;
 import com.sun.jdi.InvalidTypeException;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class Scheduler implements Runnable {
     private final DMA_Transmitter transmitterToFloor;
     private final DMA_Transmitter transmitterToElevator;
     private final DMA_Receiver receiver;
+    private List<DestinationEvent> destinationEvents;
 
     public Scheduler(DMA_Receiver receiver, DMA_Transmitter transmitterToFloor, DMA_Transmitter transmitterToElevator) {
         this.receiver = receiver;
         this.transmitterToElevator = transmitterToElevator;
         this.transmitterToFloor = transmitterToFloor;
+        destinationEvents = new ArrayList<DestinationEvent>();
     }
 
     /**
-     * process event received from the floor (up or down request)
-     * and sends it to the elevator.
-     * @param destinationEvent a destination event
+     * Process event received from the floor.
+     *
+     * @param destinationEvent Destination event received from floor.
      */
-    private void processDestinationEvent(DestinationEvent destinationEvent) {
-        transmitterToElevator.send(destinationEvent);
+    public void processDestinationEvent(DestinationEvent destinationEvent) {
+        // Store event locally to use in scheduling
+        destinationEvents.add(destinationEvent);
 
+        // Forward event to elevator subsystem
+        // TODO(@alex): describe WHY event is being forwarded to elevator subsystem
+        //   HEY ALEX! It would be helpful for the team for you to document HERE why the
+        //   elevator subsystem is getting a duplicated event. Much love, Braeden.
+        transmitterToElevator.send(destinationEvent);
     }
 
-    private boolean isElevatorStopping(ElevatorStateEvent event) {
-        // TODO: Implement check
-        return true;
+    /**
+     * Check if elevator should stop.
+     *
+     * @param e Elevator state to check.
+     * @return True if elevator should stop, false otherwise.
+     */
+    public boolean isElevatorStopping(ElevatorStateEvent e) {
+        // Elevator should stop if elevator.currentFloor belongs to the
+        // union of scheduler.destinationEvents and elevator.destinationEvents.
+        //
+        // See state diagram of scheduler for additional context.
+
+        List<DestinationEvent> union = unionOfDestinationEvents(destinationEvents, e.destinationEvents());
+        List<Integer> destinationFloors = toDestinationFloors(union);
```

Comment:
Set I believe are more efficient, and more practical for our usecases.

Reply:
Implemented as sets. Resolved.